src/main/java/com/vozniuk/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package com.vozniuk.bookstore.service.impl;
+
+import com.vozniuk.bookstore.dto.order.OrderItemDto;
+import com.vozniuk.bookstore.dto.order.OrderRequestDto;
+import com.vozniuk.bookstore.dto.order.OrderResponseDto;
+import com.vozniuk.bookstore.dto.order.OrderUpdateRequestDto;
+import com.vozniuk.bookstore.exception.EntityNotFoundException;
+import com.vozniuk.bookstore.mapper.OrderItemMapper;
+import com.vozniuk.bookstore.mapper.OrderMapper;
+import com.vozniuk.bookstore.model.CartItem;
+import com.vozniuk.bookstore.model.Order;
+import com.vozniuk.bookstore.model.OrderItem;
+import com.vozniuk.bookstore.model.ShoppingCart;
+import com.vozniuk.bookstore.repository.OrderItemRepository;
+import com.vozniuk.bookstore.repository.OrderRepository;
+import com.vozniuk.bookstore.repository.ShoppingCartRepository;
+import com.vozniuk.bookstore.repository.UserRepository;
+import com.vozniuk.bookstore.security.AuthenticationService;
+import com.vozniuk.bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final AuthenticationService authenticationService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public OrderResponseDto save(OrderRequestDto requestDto) {
+        Order order = new Order();
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        ShoppingCart shoppingCart = shoppingCartRepository
```

Comment:
Is it possible to move logic of retrieving shopping cart to shoppingCartService and call shoppingCartService  method here?

Reply:
done